rio_tiler/tasks.py

```diff
@@ -47,7 +51,10 @@ def create_tasks(
         self, reader: Callable, assets, *args, **kwargs
     ) -> Generator[Tuple[Callable, str], None, None]:
         """Create Future Tasks."""
-        return ((reader(asset, *args, **kwargs), asset) for asset in assets)
+        return (
+            (reader(asset, *args, **self._filter_kwargs(kwargs)), asset)
```

Comment:
Instead of filtering out a `threads` key from `kwargs`, would it be easier to just add `threads` as an argument to `create_tasks`?